web/src/app/pages/main-page-container/main-page/map/map.component.ts

```diff
@@ -137,52 +138,56 @@ export class MapComponent implements AfterViewInit, OnChanges, OnDestroy {
         this.activeSurvey$,
         this.lois$,
         this.navigationService.getLocationOfInterestId$(),
+        this.navigationService.getTaskId$(),
         this.selectedJob$,
-      ]).subscribe(([survey, lois, locationOfInterestId, selectedJob]) => {
-        const loisMap = ImmutableMap(
-          lois
-            .filter(loi =>
-              this.showPredefinedLoisOnly ? loi.predefined : true
+      ]).subscribe(
```

Comment:
This method is becoming unwieldy. While you're here any chance you could refactor it a bit, perhaps pulling out meaningful logic into helper functions?